### PR TITLE
GH #184: dsFPD  get led state test step update

### PR DESCRIPTION
**Pull request by akhilbhas**
update test_l1_dsFPD_positive_dsFPGetLEDState to update step 05 with comparing the value set on step 03

### CODE REVIEW

File: src/test_l1_dsFPD.c [hunk 1]
```diff
@@ -3080,7 +3080,7 @@ void test_l1_dsFPD_negative_dsFPSetLEDState(void)
  * |02|Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()|eIndicator: [Valid Indicator], state: dsFPD_STATE_ON |dsERR_NONE|Ensure the system is initialized|
  * |03|Call dsFPSetLEDState() with valid value|state: dsFPD_LED_DEVICE_ACTIVE|dsERR_NONE|The LED state should be set successfully|
  * |04|Call dsFPGetLEDState() with a valid parameter|state: dsFPDLedState_t*|dsERR_NONE|API should retrieve the current state successfully|
- * |05|Compare the retrieved LED state with the expected value from the profile file
+ * |05|Compare the retrieved LED state with the with the set value in step 03
```
Comment (hari22yuva) [inline src/test_l1_dsFPD.c:3083]: this L1 test case we need to use only dsFPGetLEDState() and it's @pre condition API's only used to test 
please update the test case like that and please check the other test case also

Comment (akhilbhas) [reply src/test_l1_dsFPD.c:3041]: Checked and updated wherever applicable

File: src/test_l1_dsFPD.c [hunk 2]
```diff
@@ -486,22 +486,20 @@ void test_l1_dsFPD_positive_dsSetFPBlink (void)
...
 
     // Step 03: Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()
     // Step 04: Call dsSetFPBlink() and loop through all valid indicators
-    // Step 05: Set all valid indicators to dsFPD_STATE_OFF using dsSetFPState()
     for (int i = 1; i <= count; i++)
     {
         snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
         eIndicator =(dsFPDIndicator_t)UT_KVP_PROFILE_GET_UINT32(buffer);
         UT_LOG("\n In %s, Indicator: [%d]\n", __FUNCTION__, eIndicator);
-        result = dsSetFPState(eIndicator, dsFPD_STATE_ON);
```
Comment (ishthiyaqahmed) [inline src/test_l1_dsFPD.c:498]: Please update the test procedures as per the changes.

File: src/test_l1_dsFPD.c [hunk 3]
```diff
@@ -486,22 +486,20 @@ void test_l1_dsFPD_positive_dsSetFPBlink (void)
 
     // Step 02: Retrieve the number of indicators from the profile file
     count = UT_KVP_PROFILE_GET_UINT8("dsFPD/Number_of_Indicators");
+    enableFPDIndicators(count);
```
Comment (ishthiyaqahmed) [inline src/test_l1_dsFPD.c:489]: this should be step 03

Comment (akhilbhas) [reply src/test_l1_dsFPD.c:490]: updated

File: src/test_l1_dsFPD.c (hunk 2 above)
Comment (akhilbhas) [reply src/test_l1_dsFPD.c:498]: updated

File: src/test_l1_dsFPD.c (hunk 1 above)
Comment (hari22yuva) [reply src/test_l1_dsFPD.c:3041]: is there any requirement to enableFPDIndicators() & disableFPDIndicators() even there is no @pre for that API?

File: src/test_l1_dsFPD.c [hunk 4]
```diff
@@ -881,8 +860,17 @@ void test_l1_dsFPD_positive_dsGetFPBrightness (void)
```
Comment (hari22yuva) [inline src/test_l1_dsFPD.c:860]: can we check with mini & max values which is defined in the profile file

File: src/test_l1_dsFPD.c [hunk 5]
```diff
@@ -1041,10 +1007,7 @@ void test_l1_dsFPD_positive_dsGetFPState (void)
         UT_ASSERT_EQUAL(result, dsERR_NONE);
     }
```
Comment (hari22yuva) [inline src/test_l1_dsFPD.c:1009]: do we need compare current state by calling again getfpstate like other test cases ?

**Review by hari22yuva — COMMENTED** [review]
Please address the review comments

File: src/test_l1_dsFPD.c (hunk 1 above)
Comment (akhilbhas) [reply src/test_l1_dsFPD.c]: No enable/disable for GetLEDState or SetLEDState implemented as it is not a prerequesite.

File: src/test_l1_dsFPD.c [hunk 6]
```diff
@@ -466,7 +466,8 @@ void test_l1_dsFPD_negative_dsSetFPState (void)
  * |03|Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()|eIndicator: [Valid Indicator], state: dsFPD_STATE_ON |dsERR_NONE|Ensure the system is initialized|
  * |04|Call dsSetFPBlink() and loop through all valid indicators from the profile file | eIndicator: [Valid Indicator], uBlinkDuration: 500, uBlinkIterations: 10 |dsERR_NONE|Should validate that the blink setup works correctly|
  * |05|Set all valid indicators to dsFPD_STATE_OFF using dsSetFPState()|eIndicator: [Valid Indicator], state: dsFPD_STATE_OFF |dsERR_OPERATION_NOT_SUPPORTED|Validate invalid parameter handling for eIndicator|
- * |06|Terminate with dsFPTerm()| |dsERR_NONE|Ensure the system is terminated|
+ * |06|Call dsSetFPBlink() and loop through all valid indicators from the profile file |eIndicator: [Valid Indicator], uBlinkDuration: 500, uBlinkIterations: 10 |dsERR_OPERATION_NOT_SUPPORTED|Testing if function adheres to pre-condition of FP State being “OFF”|
+ * |07|Terminate with dsFPTerm()| |dsERR_NONE|Ensure the system is terminated|
  *
  * @note Valid indicators can retrieved from yaml file
  *
```
[the remark this answers is not in the document]
Comment (akhilbhas) [reply src/test_l1_dsFPD.c:468]: updated

File: src/test_l1_dsFPD.c [hunk 7]
```diff
@@ -466,7 +466,8 @@ void test_l1_dsFPD_negative_dsSetFPState (void)
  * |03|Set all valid indicators to dsFPD_STATE_ON using dsSetFPState()|eIndicator: [Valid Indicator], state: dsFPD_STATE_ON |dsERR_NONE|Ensure the system is initialized|
  * |04|Call dsSetFPBlink() and loop through all valid indicators from the profile file | eIndicator: [Valid Indicator], uBlinkDuration: 500, uBlinkIterations: 10 |dsERR_NONE|Should validate that the blink setup works correctly|
  * |05|Set all valid indicators to dsFPD_STATE_OFF using dsSetFPState()|eIndicator: [Valid Indicator], state: dsFPD_STATE_OFF |dsERR_OPERATION_NOT_SUPPORTED|Validate invalid parameter handling for eIndicator|
- * |06|Terminate with dsFPTerm()| |dsERR_NONE|Ensure the system is terminated|
+ * |06|Call dsSetFPBlink() and loop through all valid indicators from the profile file |eIndicator: [Valid Indicator], uBlinkDuration: 500, uBlinkIterations: 10 |dsERR_OPERATION_NOT_SUPPORTED|Testing if function adheres to pre-condition of FP State being “OFF”|
+ * |07|Terminate with dsFPTerm()| |dsERR_NONE|Ensure the system is terminated|
```
[the remark this answers is not in the document]
Comment (akhilbhas) [reply src/test_l1_dsFPD.c:465]: updated

File: src/test_l1_dsFPD.c (hunk 5 above)
Comment (akhilbhas) [reply src/test_l1_dsFPD.c:999]: updated

File: src/test_l1_dsFPD.c (hunk 4 above)
Comment (akhilbhas) [reply src/test_l1_dsFPD.c]: updated

**Review by thanushreevani — APPROVED** [review]

**Review by hari22yuva — APPROVED** [review]

**Pull request merged**